src/mds/MDSRank.h

```diff
@@ -307,6 +307,12 @@ class MDSRank {
       std::copy(v.begin(), v.end(), std::back_inserter(finished_queue));
       progress_thread.signal();
     }
+    void queue_waiters_front(MDSInternalContextBase::vec& ls) {
+      MDSInternalContextBase::vec v;
+      v.swap(ls);
+      std::copy(v.rbegin(), v.rend(), std::front_inserter(finished_queue));
```

Comment:
Wow. Okay this needs a comment explaining what situation this is addressing. You're basically reversing the context vector and putting that at the front of the deque? This is not what one glancing at `queue_waiters_front` would expect and it's easy to miss!

Reply:
contexts in 'ls' do not get reversed.  If 'ls' is list, 'std::copy(v.rbegin(), v.rend(), std::front_inserter(finished_queue))' is equivalent to ’finished_queue.splice(finished_queue.begin(), ls)'

Reply:
OKay got it